### PR TITLE
feat: implement SIP-1.4 registration state machine

### PR DESCRIPTION
**Pull request by pcamarajr**
## Summary

Implements SIP-1.4: Registration State Machine to manage SIP account registration lifecycle. This adds a domain entity for registration state with strict state transition validation and an AuthService that orchestrates registration using existing infrastructure.

## Changes

- **Domain Entity**: Created `Registration` entity with state enum (Unregistered, Registering, Registered, Failed, Expired) and state transition validation
- **Auth Service**: Implemented `AuthService` that orchestrates SIP registration using existing `register_with_challenge` infrastructure
- **State Management**: Thread-safe registration state management with `Arc<RwLock<Registration>>`
- **Expiration Handling**: Automatic expiration checking and refresh capability
- **Module Structure**: Added services layer to project structure

### How to Test

1. **Setup**:
   ```bash
   git checkout feat/sip-1-4-registration-state-machine
   cd src-tauri
   cargo test --lib
   ```

2. **Test Steps**:
   - Run unit tests: `cargo test --lib domain::entities::registration`
   - Run service tests: `cargo test --lib services::auth_service`
   - Verify all 156 tests pass
   - Check compilation: `cargo check`
   - Verify linting: `cargo clippy --all-targets --all-features -- -D warnings`

3. **Expected Behavior**:
   - All state transitions work correctly
   - Invalid transitions are rejected with appropriate errors
   - Expiration checking works correctly
   - Service methods properly update registration state

### Test Coverage

- [x] Unit tests added/updated (17 tests for Registration entity, 7 tests for AuthService)
- [x] Integration tests added/updated (service integration with infrastructure)
- [x] Manual testing completed (all tests pass)
- [x] Edge cases tested (invalid transitions, expiration logic)

### Code Quality

- [x] Code follows project conventions
- [x] Functions are focused and single-purpose
- [x] Variable names are descriptive
- [x] No code duplication
- [x] Error handling is appropriate

### Framework Compliance

- [x] Rust code follows idiomatic patterns
- [x] Tauri IPC follows best practices (services layer ready for commands)
- [x] Design system components used correctly (N/A for backend)

### Security

- [x] No hardcoded secrets
- [x] Input validation present (credentials validation)
- [x] Sensitive data handled securely (credentials stored in entity)
- [x] Tauri capabilities properly configured (N/A for this change)

### Documentation

- [x] README.md updated (Phase 4 section updated)
- [x] Code comments explain complex logic (state transitions documented)
- [x] Architecture docs updated (roadmap marked SIP-1.4 as completed)

## Breaking Changes

None - This is a new feature that adds functionality without breaking existing code.

## Related Issues

Implements SIP-1.4 from the implementation roadmap (Phase 4).

## Additional Notes

- The `AuthService` is ready for integration with Tauri commands (SIP-1.5)
- Expiration monitoring can be started with `start_expiration_monitor()` method
- All state transitions are validated to prevent invalid state changes
- The service integrates seamlessly with existing `register_with_challenge` infrastructure

### CODE REVIEW

File: src-tauri/src/domain/entities/registration.rs [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+    pub fn last_error(&self) -> Option<&String> {
+        self.last_error.as_ref()
+    }
+
+    /// Check if registration is expired
+    pub fn is_expired(&self) -> bool {
+        if let Some(expires_at) = self.expires_at {
+            let now = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
```
Comment (pcamarajr) [inline src-tauri/src/domain/entities/registration.rs]: **🚫 Blocker:** Uses `unwrap()` which can panic

`SystemTime::now().duration_since(UNIX_EPOCH).unwrap()` can panic if the system time is before the Unix epoch (unlikely but possible). This should use proper error handling. However, since `is_expired()` returns `bool`, consider handling the error gracefully by returning `false` if the system time is invalid.

```rust
// ❌ Current
let now = SystemTime::now()
    .duration_since(UNIX_EPOCH)
    .unwrap()
    .as_secs();

// ✅ Should be
let now = match SystemTime::now().duration_since(UNIX_EPOCH) {
    Ok(duration) => duration.as_secs(),
    Err(_) => return false, // If system time is invalid, consider not expired
};
```

File: src-tauri/src/services/auth_service.rs [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    /// * `check_interval_secs` - How often to check for expiration (default: 60 seconds)
+    ///
+    /// # Returns
+    /// A handle to stop the background task
+    pub fn start_expiration_monitor(
+        &self,
+        check_interval_secs: u64,
+    ) -> tokio::task::JoinHandle<()> {
+        let registration = Arc::clone(&self.registration);
+        let mut service = AuthService {
```
Comment (pcamarajr) [inline src-tauri/src/services/auth_service.rs]: **⚠️ Warning:** Creates new AuthService instance unnecessarily

The `start_expiration_monitor` method creates a new `AuthService` instance just to call `refresh_registration()`. This is inefficient and could lead to state synchronization issues. Consider refactoring to either:

1. Extract `refresh_registration` logic into a helper that doesn't require `&mut self`
2. Store the necessary fields (`server_addr`, `contact_uri`) in the closure directly and call the registration infrastructure function directly
3. Use `Arc<AuthService>` if the service needs to be shared

```rust
// Current approach creates unnecessary instance
let mut service = AuthService {
    registration: Arc::clone(&self.registration),
    client: Arc::clone(&self.client),
    server_addr: self.server_addr,
    contact_uri: self.contact_uri.clone(),
};
```

**Review by pcamarajr — COMMENTED** [review]
## Code Review Summary

✅ **Overall Assessment:** The PR implements SIP-1.4 registration state machine correctly with good architecture compliance. However, there are critical issues that must be addressed before merging.

**Issues Found:**
- 🚫 Blockers: 1
- ⚠️ Warnings: 1
- 💡 Suggestions: 0

**Key Findings:**
- `unwrap()` usage in `is_expired()` method can cause panics (blocker)
- Unnecessary `AuthService` instance creation in `start_expiration_monitor` (warning)

**Review Details:**
- Inline comments posted for line-specific issues
- Category review comments posted above for categories with issues
- Categories reviewed: Functionality & Correctness, Code Quality & Maintainability, Security, Architecture Compliance, Rust & Tauri Compliance, Testing & Documentation, Performance & Optimization, README & Documentation Updates

**Positive Aspects:**
- ✅ Excellent architecture compliance - follows Clean Architecture layers correctly
- ✅ Comprehensive test coverage (17 tests for Registration entity, 7 tests for AuthService)
- ✅ Proper state machine implementation with validation
- ✅ README and documentation properly updated
- ✅ Good error handling in most places (except the `unwrap()` issue)
- ✅ Thread-safe state management with `Arc<RwLock<>>`
- ✅ Proper integration with existing infrastructure

**Required Actions:**
1. Fix the `unwrap()` in `is_expired()` method (blocker)
2. Consider refactoring `start_expiration_monitor` to avoid creating new service instances (warning)

File: src-tauri/src/domain/entities/registration.rs (hunk 1 above)
Comment (pcamarajr) [reply src-tauri/src/domain/entities/registration.rs]: ✅ Resolved in commit c77bce56c38ee47ee535c52bc9bd6772c0b59b75
